packages/beacon-node/src/chain/ColumnReconstructionTracker.ts

```diff
@@ -63,10 +63,14 @@ export class ColumnReconstructionTracker {
     this.lastBlockRootHex = blockInput.blockRootHex;
     const delay =
       RECONSTRUCTION_DELAY_MIN_MS + Math.random() * (RECONSTRUCTION_DELAY_MAX_MS - RECONSTRUCTION_DELAY_MIN_MS);
-    sleep(delay).then(() => {
-      recoverDataColumnSidecars(blockInput, this.emitter, this.metrics).finally(() => {
-        this.running = false;
+    sleep(delay)
+      .then(() => {
+        recoverDataColumnSidecars(blockInput, this.emitter, this.metrics).finally(() => {
+          this.running = false;
+        });
+      })
+      .catch((err) => {
+        this.logger.debug("ColumnReconstructionTracker occurred an error", err);
```

Comment:
```suggestion
      .catch((err) => {
        this.logger.debug("ColumnReconstructionTracker occurred an error", {}, err);
```
also wouldn't we want ot set `this.running = false;` in that case too?

Reply:
ok, the placement of this catch is weird.
`sleep` only throws if an abort signal is passed in. (It isn't passed in, so it won't throw)
So this catch is meant to catch an promise rejection of `recoverDataColumnSidecars`.

In that case, the `catch` should be attached to recoverDataColumnSidecars directly.

> we want ot set this.running = false

that's already covered by the `finally`

Reply:
If we add `catch` to the `recoverDataColumnSidecars` the return value for `sleep` will still be a floating promise which is not handled.